File: src/main/java/info/idgst/user/UserRepository.java
package info.idgst.user;

import org.springframework.data.repository.CrudRepository;

/**
 * Repository for retrieving Users.
 *
 * @author Aliaksei Bahdanau
 */
public interface UserRepository extends CrudRepository<AppUser, String> {

    AppUser findByUserName(String userName);
}

File: src/main/kotlin/info/devbug/user/UserRepository.kt
package info.devbug.user

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

/**
 * @author Aliaksei Bahdanau
 */
@Repository
interface UserRepository : CrudRepository<UserDto, String> {

    fun findByUserName(userName: String): UserDto
}